recipe/meta.yaml

```diff
@@ -1,52 +1,82 @@
+# NB: this recipe should be synced between sklearnex repo and feedstocks
+
 {% set name = "scikit-learn-intelex" %}
-{% set version = "2024.6.0" %}
-{% set buildnumber = 1 %}
+{% set version = "2024.7.0" %}
+{% set buildnumber = 0 %}
 
 package:
   name: {{ name|lower }}
   version: {{ version }}
 
 source:
   url: https://github.com/intel/scikit-learn-intelex/archive/{{ version }}.tar.gz
-  sha256: 7ae556ca5923d3fb7a88f6534f4c48984bc8616b3a3827d7d44b655c6b92624b
+  sha256: 76371955787a30fd7ef8563a8083555cdc74841711a393b0a9194fb326c28acd
 
 build:
   skip: true  # [not (linux64 or win)]
   number: {{ buildnumber }}
-  script:
-    - export DALROOT="${PREFIX}"  # [not win]
-    - set DALROOT=%PREFIX%        # [win]
-    - {{ PYTHON }} setup_sklearnex.py install --single-version-externally-managed --record=record_sklearnex.txt
+  include_recipe: False
+  script_env:
+  - DPCPPROOT
+  - DALROOT
+  - NO_DIST=1  # [win]
 
 requirements:
+  build:
+    - make  # [linux]
+    - dpcpp_linux-64  # [linux64]
+    # - dpcpp_win-64  # [win]
+    - {{ compiler('cxx') }}  # [linux64 or win]
+    # conda-forge feedstock specific
+    - {{ stdlib("c") }}  # [linux64 or win]
   host:
     - python
     - setuptools
-    - dal-include =={{ version }}
+    - cmake
+    - clang-format
+    - cython
+    - jinja2
+    - pybind11
+    - numpy {{ numpy }}
+    - mpich  # [not win]
+    # - impi-devel  # [win]
+    # dal-devel pinning depends on the recipe location (repo or feedstock)
+    # - dal-devel
+    - dal-devel =={{ version }}
   run:
     - python
-    - daal4py =={{ version }}
-    - scikit-learn
+    - {{ pin_compatible('numpy') }}
+    - dpcpp-cpp-rt  # [linux64]
+    # dal pinning depends on the recipe location (repo or feedstock)
+    # - dal
+    - dal =={{ version }}
 
 test:
   requires:
     - pandas
+    - scikit-learn
+    - xgboost
+    - lightgbm
+    - catboost
```

Comment:
This would complicate any migration as we will be forced to wait for those first to complete migration. this might happen much later than we actually can migrate.

Reply:
There is no significant problem to temp. remove testing dependencies like it's done for dpctl, dpnp, shap and catboost.